--- a/make.tmpl.in
+++ b/make.tmpl.in
@@ -35,7 +35,7 @@
 INSTALL_DATA=${INSTALL} -m 644
 
 LDFLAGS=@LDFLAGS@ -Wl,--as-needed
-CXXFLAGS= -D@GUI@ $(local_inc) -I$(top_srcdir)/include/base -I$(top_srcdir)/include/x @x_includes@ @gui_includes@ @CXXFLAGS@
+CXXFLAGS= -D@GUI@ $(local_inc) -I$(top_srcdir)/include/base -I$(top_srcdir)/include/x @x_includes@ @gui_includes@ -Wno-narrowing -fpermissive @CXXFLAGS@
 CFLAGS= -D@GUI@ $(local_inc) -I$(top_srcdir)/include/base -I$(top_srcdir)/include/x @x_includes@ @gui_includes@  @CFLAGS@
 PICFLAGS= @PICFLAGS@
 
