--- a/Makefile.in
+++ b/Makefile.in
@@ -71,6 +71,7 @@
 	   || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
 	done && test -z "$$fail"
 	rm -f config.cache config.status config.log include/WXconfig.h
+	rm -rf bin
 
 distclean:
 	for i in ${instdirs}; do \
@@ -78,8 +79,9 @@
 	  (cd $$i; ${MAKE} distclean) \
 	   || case "$(MFLAGS)" in *k*) fail=yes;; *) exit 1;; esac; \
 	done && test -z "$$fail"
-	rm config.cache config.status config.log include/WXconfig.h
-	find . -name Makefile -exec rm {} \; -print
+	rm -f config.cache config.status config.log include/WXconfig.h
+	rm -f make.tmpl
+	find . -name Makefile -exec rm -f {} \; -print
 
 dist:
 	tar cvf wxshlib1.67-1.tar config configure config.guess config.sub\
--- a/utils/prologio/src/Makefile.in
+++ b/utils/prologio/src/Makefile.in
@@ -64,6 +64,7 @@
 install::
 
 clean::
+	rm -f lex_yy.c y_tab.c
 	rm -f $(OBJS) $(DEMOOBJS) $(DEMOPROGS) *~ core *.bak ${SH_OBJS}
 	rm -f $(doc_src)/*.dvi $(doc_src)/*.aux $(doc_src)/*..bb\
 	$(doc_src)/*.blg $(doc_src)/*.idx $(doc_src)/*.ilg $(doc_src)/*.ind\
--- a/utils/tex2rtf/src/wxwin/Makefile.in
+++ b/utils/tex2rtf/src/wxwin/Makefile.in
@@ -79,6 +79,7 @@
 html::	$(HTMLNAME)
 
 clean::
+	rm -f ${TEX2RTFSRCS}
 	rm -f $(OBJS) $(DEMOOBJS) $(DEMOPROGS) *~ core *.bak ${SH_OBJS}
 	rm -f $(doc_src)/*.dvi $(doc_src)/*.aux $(doc_src)/*..bb\
 	$(doc_src)/*.blg $(doc_src)/*.idx $(doc_src)/*.ilg $(doc_src)/*.ind\
